xugrid/ugrid/partitioning.py

```diff
@@ -328,6 +328,8 @@ def merge_partitions(partitions, merge_ugrid_chunks: bool = True):
     -------
     merged : UgridDataset
     """
+    if len(partitions) == 0:
+        raise ValueError("Received empty partitions list, cannot be merged.")
```

Comment:
Pedantic: partitions is a sequence, so something like a tuple is allowed too. Maybe:

```python
ValueError("Cannot merge partitions: partitions is empty")
```

What I dislike is that this might give a user the idea that the partitions are maybe empty datasets.

So maybe:

```python
ValueError("Cannot merge partitions: zero partitions provided")
```

While we're at it: Also makes sense to run partitions[0] if the length is 1, right?

I just checked what xarray does for a merge on an empty list: it returns an empty dataset:

```python
In [1]: import xarray as xr

In [2]: xr.merge([])
Out[2]:
<xarray.Dataset> Size: 0B
Dimensions:  ()
Data variables:
    *empty*
```

This probably isn't what we want, merging partitions is quite specific, shows a clear intent to really do something.

Reply:
Message was updated to `"Cannot merge partitions: zero partitions provided"` and testbank is green again, the `len(partitions)` failed some testcases so I replaced it with `len(types)`.